storybook/src/components/button/button.css-template.ts

```diff
@@ -8,24 +8,30 @@ export const cssButton: ComponentImplementation<Button | ButtonAnchor> = {
   component: "button",
   implementation: "html-css",
   template: ({ iconTemplate, tooltipTemplate }) => {
-    function getClassName(variant: "primary" | "secondary" | "tertiary" | null, modifier: string | undefined) {
-      if (variant && modifier) {
-        return `dso-${variant} ${modifier}`;
-      }
+    function getClassName(
+      variant: "primary" | "secondary" | "tertiary" | null,
+      modifier: string | undefined,
+      mode?: string | undefined
+    ) {
+      let classes = "";
 
       if (variant) {
-        return `dso-${variant}`;
+        classes += `dso-${variant}`;
+      }
+
+      if (mode) {
+        classes += variant ? ` ${mode}` : mode;
       }
 
       if (modifier) {
-        return modifier;
+        classes += variant || mode ? ` ${modifier}` : modifier;
```

Comment:
Met `const classes = [];` en elke class op deze array te pushen voorkom je dit soort testjes om de spatie op de juiste plek te krijgen.

Je kan aan het einde het aantal classes tellen voor undefined, of anders undefined teruggeven:

* `return classes.length > 0 ? classes.join(' ') : undefined;`
* `return classes.join(' ') || "";`

Maar de stringbuilder is mijn minst favoriete variant. Foutgevoelig en moeilijk onderhoudbaar/uitbreidbaar.

Reply:
Dat is inderdaad mooier, ik had dit even snel uit de mouw geschud.